File: src/types/homepage.ts
export interface Internship {
  id: number
  company: string
  dept: string
  role: string
  period: string
  contribution: string
  photoUrl?: string
}

export interface StarItem {
  label: 'S' | 'T' | 'A' | 'R'
  text: string
}

export type ProjectType = 'code' | 'uiux' | 'finance'

export interface Project {
  id: number
  name: string
  type: string // comma-separated ProjectType values, e.g. "code,finance"
  techLabel: string
  tech: string
  members: number
  period: string
  core: string
  githubUrl?: string
  youtubeUrl?: string
  otherUrl?: string
  star: StarItem[]
  createdAt: string
}

export interface LangCert {
  name: string
  score: string
  pct: number
}

export interface LangCertAdmin {
  id: number
  lang: 'en' | 'jp'
  name: string
  score: string
  pct: number
}

export interface CertGroup {
  category: string
  items: string[]
}

export interface CertGroupAdmin {
  id: number
  domain: '財會' | '資訊' | 'finance' | 'it'
  category: string
  items: string[]
  sortOrder: number
}

export interface CertData {
  language: { en: LangCert[]; jp: LangCert[] }
  finance: CertGroup[]
  it: CertGroup[]
}

export interface AcademicMilestone {
  id: number
  school: string
  major: string
  period: string
  gpa: string
  rank: string
  facts: string[]
  sortOrder?: number
}

export type PlanPhase = 'short' | 'mid-short' | 'mid'

export interface FuturePlan {
  id: number
  phase: PlanPhase
  title: string
  subtitle: string
  items: string[]
  sortOrder?: number
}

File: backend/app/models/homepage.py
from sqlalchemy import Column, Integer, String, Text, Float, Date
from ..database import Base


class Internship(Base):
    __tablename__ = "internships"
    id           = Column(Integer, primary_key=True)
    company      = Column(String(200), nullable=False)
    dept         = Column(String(200), nullable=False, default="")
    role         = Column(String(100), nullable=False, default="實習生")
    start_date   = Column(Date, nullable=True)
    end_date     = Column(Date, nullable=True)
    contribution = Column(Text, nullable=False)
    photo_url    = Column(String(500), nullable=True)


class Project(Base):
    __tablename__ = "projects"
    id          = Column(Integer, primary_key=True)
    name        = Column(String(200), nullable=False)
    type        = Column(String(20), nullable=False)   # comma-separated subset of code|uiux|finance
    tech_label  = Column(String(100), nullable=False, default="主要職責")
    tech        = Column(String(500), nullable=False, default="")
    members     = Column(Integer, default=1)
    period      = Column(String(50), nullable=False, default="")
    core        = Column(String(200), nullable=False, default="")
    github_url  = Column(String(500), nullable=True)
    youtube_url = Column(String(500), nullable=True)
    other_url   = Column(String(500), nullable=True)
    star        = Column(Text, default="[]")           # JSON: [{label, text}, ...]
    created_at  = Column(String(20), nullable=False, default="")


class LanguageCert(Base):
    __tablename__ = "language_certs"
    id    = Column(Integer, primary_key=True)
    lang  = Column(String(10), nullable=False)          # 'en' | 'jp'
    name  = Column(String(100), nullable=False)
    score = Column(String(50), nullable=False)
    pct   = Column(Float, nullable=False, default=0.0)


class CertGroup(Base):
    __tablename__ = "cert_groups"
    id         = Column(Integer, primary_key=True)
    domain     = Column(String(20), nullable=False)     # 'finance' | 'it'
    category   = Column(String(100), nullable=False)
    items      = Column(Text, default="[]")             # JSON: list[str]
    sort_order = Column(Integer, default=0)


class AcademicMilestone(Base):
    __tablename__ = "academic_milestones"
    id         = Column(Integer, primary_key=True)
    school     = Column(String(200), nullable=False)
    major      = Column(String(200), nullable=False)
    period     = Column(String(50), nullable=False)
    gpa        = Column(String(50), nullable=False, default="")
    rank       = Column(String(50), nullable=False, default="")
    facts      = Column(Text, default="[]")             # JSON: list[str]
    sort_order = Column(Integer, default=0)


class FuturePlan(Base):
    __tablename__ = "future_plans"
    id         = Column(Integer, primary_key=True)
    phase      = Column(String(20), nullable=False)     # short | mid-short | mid
    title      = Column(String(100), nullable=False)
    subtitle   = Column(String(200), nullable=False)
    items      = Column(Text, default="[]")             # JSON: list[str]
    sort_order = Column(Integer, default=0)
